src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1064,6 +1066,18 @@ internal new ModuleSymbol SourceModule
             }
         }
 
+        internal VariableUsePass VariableUsePass
+        {
+            get
+            {
+                if (_lazyVariableUsePass == null)
+                {
+                    Interlocked.CompareExchange(ref _lazyVariableUsePass, new VariableUsePass(this.SourceAssembly), null);
```

Comment:
Well scratch that, it looks like VariableUsePass accumulates some state in the instance, interlocked exchange is required then.

---
In reply to: [116881355](https://github.com/dotnet/roslyn/pull/19444#discussion_r116881355) [](ancestors = 116881355)